cdc/sinkv2/eventsink/event_appender.go

```diff
@@ -42,10 +46,50 @@ var _ Appender[*model.SingleTableTxn] = (*TxnEventAppender)(nil)
 type TxnEventAppender struct{}
 
 // Append appends the given rows to the given txn buffer.
+// The callers of this function should **make sure** that
+// the commitTs and startTs of rows is **strictly increasing**.
+// 1. Txns ordered by commitTs and startTs.
+// 2. Rows grouped by startTs, cause the StartTs is the unique identifier of the transaction.
 func (t *TxnEventAppender) Append(
 	buffer []*model.SingleTableTxn,
 	rows ...*model.RowChangedEvent,
 ) []*model.SingleTableTxn {
-	// TODO implement me
-	panic("implement me")
+	for _, row := range rows {
+		noTxn := len(buffer) == 0
+
+		if noTxn ||
```

Comment:
I wonder if anyone has any suggestions for simplifying this part of the code? I don't seem to have a simpler way to state and define these conditions.

Reply:
See: https://github.com/pingcap/tiflow/pull/6109/commits/491089b40f7b674c0498fe305a0abfd8bfe9132c